ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/LedgerApiTestTool.scala

```diff
@@ -0,0 +1,180 @@
+import scala.collection.breakOut
+
+object LedgerApiTestTool {
+  def main(args: Array[String]): Unit = {
+    implicit val system: ActorSystem = ActorSystem("LedgerApiTestTool")
+    implicit val mat: ActorMaterializer = ActorMaterializer()(system)
+    implicit val ec: ExecutionContext = mat.executionContext
+    implicit val esf: AkkaExecutionSequencerPool =
+      new AkkaExecutionSequencerPool("esf-" + this.getClass.getSimpleName)(system)
+
+    // FIXME(TW) daml-prim location has just become deprecated
+    val testResources = List("/ledger/ledger-api-integration-tests/SemanticTests.dar")
+
+    val config = argParser
+      .parse(args, defaultConfig)
+      .getOrElse(sys.exit(1))
+
+    if (config.extract) {
+      extractTestFiles(testResources)
+      System.exit(1)
+    }
+
+    val packages: Map[PackageId, Ast.Package] = testResources
+      .flatMap(loadAllPackagesFromResource)(breakOut)
+
+    val scenarios = SemanticTester.scenarios(packages)
+    val nScenarios: Int = scenarios.foldLeft(0)((c, xs) => c + xs._2.size)
+
+    println(s"Running ${nScenarios} scenarios against ${config.host}:${config.port}...")
+
+    val ledgerResource = RemoteServerResource(config.host, config.port)
+      .map {
+        case PlatformChannels(channel) =>
+          LedgerContext.SingleChannelContext(channel, None, packages.keys)
+      }
+    ledgerResource.setup()
+    val ledger = ledgerResource.value
+
+    if (config.performReset) {
+      Await.result(ledger.reset(), 10.seconds)
+    }
+    var failed = false
+
+    try {
+      scenarios.foreach {
+        case (pkgId, names) =>
+          val tester = new SemanticTester(
+            parties => new SemanticTestAdapter(ledger, packages, parties.map(_.underlyingString)),
+            pkgId,
+            packages)
+          names
+            .foreach { name =>
+              println(s"Testing scenario: $name")
+              val _ = try {
+                Await.result(
+                  tester.testScenario(name),
+                  10.seconds
+                )
+              } catch {
+                case (t: Throwable) =>
+                  sys.error("Timed-out waiting for an expected event: " + t.getMessage)
+              }
+            }
+      }
+      println("All scenarios completed.")
+    } catch {
+      case (t: Throwable) => {
+        failed = true
+        if (!config.mustFail) throw t
+      }
+    } finally {
+      ledgerResource.close()
+      mat.shutdown()
+      val _ = Await.result(system.terminate(), 5.seconds)
+    }
+
+    if (config.mustFail) {
+      if (failed) println("One or more scenarios failed as expected.")
+      else
+        throw new RuntimeException(
+          "None of the scenarios failed, yet the --must-fail flag was specified!")
+    }
+  }
+
+  private def loadAllPackagesFromResource(resource: String): Map[PackageId, Ast.Package] = {
+    // TODO: replace with stream-supporting functions from UniversalArchiveReader when
```

Comment:
Please create an issue in backlog keeping track of this: https://github.com/digital-asset/daml/milestone/6

and add issue reference to code.

Reply:
It's already referenced in the next line.